### PR TITLE
feat(cli): import from crds.dev

### LINKED ISSUE #377: add crds.dev import shortcut like `cdk8s import crossplane/crossplane@0.14.0`
https://doc.crds.dev scans any repo for CRDs and can output a single manifest file [with a new `raw/` prefix](https://twitter.com/prasek/status/1328124155678527488) that @hasheddan added, so you can use: 
`cdk8s import -l typescript https://doc.crds.dev/raw/github.com/crossplane/crossplane@v0.14.0`.

@eladb suggested on [twitter](https://twitter.com/emeshbi/status/1328130666706460673):
> We should add http://crds.dev import shortcut like `cdk8s import crossplane/crossplane@0.14.0`

Also, based on some early work in https://github.com/eladb/crossplane-cdk/pull/4 it would be nice to have an `index.ts` generated for each package, so you can do an L1 import statement like this:

```ts
 import { CompositeResourceDefinition, Composition } from './imports/apiextensions.crossplane.io';
```

from this:
```
src/imports
├── apiextensions.crossplane.io
│   ├── compositeresourcedefinition.ts
│   ├── composition.ts
│   └── index.ts
├── meta.pkg.crossplane.io
│   ├── configuration.ts
│   ├── index.ts
│   └── provider.ts
```

and the providers, e.g. `provider-aws`:
```
├── acm.aws.crossplane.io
│   └── certificate.ts
├── acmpca.aws.crossplane.io
│   ├── certificateauthority.ts
│   └── certificateauthoritypermission.ts
├── aws.crossplane.io
│   ├── provider.ts
│   ├── providerconfig.ts
│   └── providerconfigusage.ts
├── cache.aws.crossplane.io
│   ├── cachecluster.ts
│   ├── cachesubnetgroup.ts
│   └── replicationgroup.ts
├── database.aws.crossplane.io
│   ├── dbsubnetgroup.ts
│   ├── dynamotable.ts
│   └── rdsinstance.ts
├── ec2.aws.crossplane.io
│   ├── elasticip.ts
│   ├── internetgateway.ts
│   ├── natgateway.ts
│   ├── routetable.ts
│   ├── securitygroup.ts
│   ├── subnet.ts
│   └── vpc.ts
├── ecr.aws.crossplane.io
│   └── repository.ts
├── eks.aws.crossplane.io
│   ├── cluster.ts
│   └── nodegroup.ts
├── elasticloadbalancing.aws.crossplane.io
│   ├── elb.ts
│   └── elbattachment.ts
├── identity.aws.crossplane.io
│   ├── iamgroup.ts
│   ├── iamgrouppolicyattachment.ts
│   ├── iamgroupusermembership.ts
│   ├── iampolicy.ts
│   ├── iamrole.ts
│   ├── iamrolepolicyattachment.ts
│   ├── iamuser.ts
│   └── iamuserpolicyattachment.ts
├── notification.aws.crossplane.io
│   ├── snssubscription.ts
│   └── snstopic.ts
├── redshift.aws.crossplane.io
│   └── cluster.ts
├── route53.aws.crossplane.io
│   ├── hostedzone.ts
│   └── resourcerecordset.ts
├── s3.aws.crossplane.io
│   ├── bucket.ts
│   └── bucketpolicy.ts
└── sqs.aws.crossplane.io
    └── queue.ts
```

### PR DESCRIPTION
**Pull request by prasek**
Introduce crds.dev import shortcut: `cdk8s import github:account/repo[@VERSION]`

Example:
`cdk8s import github:crossplane/crossplane@0.14.0`

Resolves #377

In a future PR would like to generate a single module per k8s API group with all CRDs for that API group, so you could:
```ts
import { CompositeResourceDefinition, Composition } from './imports/apiextensions.crossplane.io';
```

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by prasek** [comment]
@eladb @iliapolo how does this look?

File: packages/cdk8s-cli/src/cli/cmds/import.ts [hunk 1]
```diff
@@ -18,6 +18,7 @@ class Command implements yargs.CommandModule {
     .example('cdk8s import k8s@1.13.0', 'Imports a specific version of the Kubernetes API')
     .example('cdk8s import jenkins.io_jenkins_crd.yaml', 'Imports constructs for the Jenkins custom resource definition from a file')
     .example('cdk8s import mattermost:=mattermost_crd.yaml', 'Imports constructs for the mattermost cluster custom resource definition using a custom module name')
+    .example('cdk8s import crossplane/crossplane@0.14.0', 'Imports constructs for a GitHub repo using doc.crds.dev')
```
Comment (eladb) [inline packages/cdk8s-cli/src/cli/cmds/import.ts:21]: Should we change to `github://crossplane/crossplane@0.14.0` (or `gh://` or `gh:`). WDYT?

Add to line 16:

```
import spec with the syntax [NAME:=]SPEC where NAME is an optional module name and supported SPEC are: k8s, crd.yaml, https://domain/crd.yaml, github://account/repo[@VERSION]
```

File: packages/cdk8s-cli/src/import/dispatch.ts [hunk 2]
```diff
@@ -24,6 +24,24 @@ async function matchImporter(importSpec: ImportSpec, argv: any) {
...
+  // crds.dev import shortcut e.g. crossplane/crossplane@0.14.0
+  const match = /^([A-Za-z0-9_.-]+)\/([A-Za-z0-9_.-]+)\@([0-9]+)\.([0-9]+)(?:\.([0-9]+))?$/.exec(importSpec.source);
+  if (match) {
+    const org = match[1];
+    const repo = match[2];
+    const major = match[3];
+    const minor = match[4];
+    const patch = match[5] ?? '0';
+    const url = `https://doc.crds.dev/raw/github.com/${org}/${repo}@v${major}.${minor}.${patch}`;
+    console.log(`downloading ${url} ...`);
```
Comment (eladb) [inline packages/cdk8s-cli/src/import/dispatch.ts:36]: Remove this, not needed.

File: packages/cdk8s-cli/test/import/fixtures/crossplane.yaml [hunk 3]
```diff
@@ -0,0 +1,969 @@
+apiVersion: apiextensions.k8s.io/v1
```
Comment (eladb) [inline packages/cdk8s-cli/test/import/fixtures/crossplane.yaml:1]: What's the rationale for adding this?

**Review by eladb — CHANGES_REQUESTED** [review]
Looks good. Feels like we should add some form of unit test?

File: packages/cdk8s-cli/src/cli/cmds/import.ts (hunk 1 above)
Comment (prasek) [reply packages/cdk8s-cli/src/cli/cmds/import.ts:21]: github:// sounds good. Will update.

**Comment by eladb** [comment]
@prasek any updates? Also, make sure to re-request a review when ready for another round.

File: packages/cdk8s-cli/src/cli/cmds/import.ts (hunk 1 above)
Comment (prasek) [reply packages/cdk8s-cli/src/cli/cmds/import.ts:21]: /cc @hasheddan WDYT? ☝️ 

I'm leaning towards the slightly shorter `cdk8s import github:crossplane/crossplane@0.14.0`

File: packages/cdk8s-cli/test/import/fixtures/crossplane.yaml (hunk 3 above)
Comment (prasek) [reply packages/cdk8s-cli/test/import/fixtures/crossplane.yaml:1]: @eladb just wanted to make sure that some key crossplane CRDs passed all the cdk8s CRD import tests.

now that I've verified that I can remove if you like or leave it to ensure they continue to pass. it does bump the unit test time by a few seconds, so happy to remove.

File: packages/cdk8s-cli/src/cli/cmds/import.ts (hunk 1 above)
Comment (eladb) [reply packages/cdk8s-cli/src/cli/cmds/import.ts:21]: `github:` is find by me

Comment (hasheddan) [reply packages/cdk8s-cli/src/cli/cmds/import.ts:21]: `github:` SGTM

File: packages/cdk8s-cli/test/import/fixtures/crossplane.yaml (hunk 3 above)
Comment (prasek) [reply packages/cdk8s-cli/test/import/fixtures/crossplane.yaml:1]: > Looks good. Feels like we should add some form of unit test?

@eladb that works ... what's your preference
a. full `import-k8s-test.ts` style test with multiple versions tested and validated against the snapshot (would increase unit test times)
b. simpler regex test similar to import-k8s-util.test.ts?

Comment (eladb) [reply packages/cdk8s-cli/test/import/fixtures/crossplane.yaml:1]: Happy to add it, but let's do it in a separate PR?

Comment (prasek) [reply packages/cdk8s-cli/test/import/fixtures/crossplane.yaml:1]: sounds good 👍

File: packages/cdk8s-cli/src/cli/cmds/import.ts (hunk 1 above)
Comment (prasek) [reply packages/cdk8s-cli/src/cli/cmds/import.ts]: fixed

File: packages/cdk8s-cli/src/import/dispatch.ts (hunk 2 above)
Comment (prasek) [reply packages/cdk8s-cli/src/import/dispatch.ts]: done

**Comment by prasek** [comment]
@eladb should be good to go, thx.

File: packages/cdk8s-cli/src/import/crds-dev.ts [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+/**
+ * Matches a https://doc.crds.dev repo
+ *
+ *  - ManifestObjectDefinition[] if found
+ *  - undefined if not
+ *
+ * @param source
+ */
+export async function importCrdsDevRepoMatch (source: string): Promise<undefined | ManifestObjectDefinition[]> {
```
Comment (eladb) [inline packages/cdk8s-cli/src/import/crds-dev.ts:24]:
```suggestion
export async function importCrdsDevRepoMatch(source: string): Promise<undefined | ManifestObjectDefinition[]> {
```

Comment (eladb) [inline packages/cdk8s-cli/src/import/crds-dev.ts:24]: Can you please add a unit test for this function under `test/crds-dev.test.ts`?

**Review by eladb — CHANGES_REQUESTED** [review]

File: packages/cdk8s-cli/src/import/crds-dev.ts (hunk 4 above)
Comment (prasek) [reply packages/cdk8s-cli/src/import/crds-dev.ts]: fixed

Comment (prasek) [reply packages/cdk8s-cli/src/import/crds-dev.ts]: Done, it's got 100% coverage.

**Comment by prasek** [comment]
@eladb test coverage is 100% should be good to go, thx.

**Review by eladb — APPROVED** [review]

**Pull request merged**